pkg/api/2018-09-30-preview/api/types.go

```diff
@@ -64,6 +64,9 @@ type Properties struct {
 
 	// AuthProfile (in): configures OpenShift authentication
 	AuthProfile *AuthProfile `json:"authProfile,omitempty"`
+
+	// AuthGroupSync (in): synchronize OpenShift groups and users from the following AAD groups
+	AuthGroupSync *AuthGroupSync `json:"authGroupSync,omitempty"`
```

Comment:
This feels like it's part of `AuthProfile` and specifically of `AADIdentityProvider`.

Reply:
I think this might be right. We might try to keep it generic for generic groupSync (github teams in example) but it looks like it is part of AADIdentityProvider

Reply:
Even if we want to keep it generic across providers, isn't it still part of `AuthProfile`?

Reply:
@jim-minter do you have an opinion on this? I am not bothered either way.

Reply:
I think it probably needs to change, but I'm not sure right now to what.

Reply:
OK, I'm with @kargakis - add fields CustomerAdminGroupID and CustomerReaderGroupID to AADIdentityProvider.
They should be pointers in the internal representation as well, in case a customer doesn't want to set them at onboarding.
Also need validation for these - nil or pointer to valid UUID.